src/System.Collections/tests/Generic/Dictionary/Dictionary.Tests.cs

```diff
@@ -126,6 +126,20 @@ public void IDictionary_NonGeneric_Contains_KeyOfWrongType(int count)
             }
         }
 
+        [Fact]
+        public void Clear_OnEmptyCollection_DoesNotInvalidateEnumerator()
+        {
+            if (ModifyEnumeratorAllowed.HasFlag(ModifyOperation.Clear))
+            {
+                IDictionary dictionary = new Dictionary<string, string>();
+                IEnumerator valuesEnum = dictionary.GetEnumerator();
+
+                dictionary.Clear();
+                Assert.Empty(dictionary);
+                valuesEnum.MoveNext();
```

Comment:
Assert the expected result of MoveNext?

Reply:
In this case `MoveNext` will always be false or thrown an `InvalidOperationException`. 

It's probably better to be explicit, but wouldn't an `Assert.False()` here be a bit confusing? I.e. whether or not the enumerator advances or not is an implementation detail of `IEnumerator` we don't necessarily want to test with this.

Reply:
> whether or not the enumerator advances or not is an implementation detail of IEnumerator we don't necessarily want to test with this.

How is it an implementation detail? Regardless of implementation, it should always return false.

Reply:
Absolutely true. 

I suppose my confusion here stems from the fact that "invalidate the enumerator" here means not changing the state of the collection (and not throwing an InvalidOperationException). The functionality of `Clear` is already covered by other tests. Whether or not `MoveNext` is false or not doesn't necessarily matter for the purpose of this test.

Reply:
Either way, I'm deferring to you - added.